base/logging_context.go

```diff
@@ -44,6 +44,8 @@ type LogContext struct {
 	// TestName can be a unit test name (see TestCtx)
 	TestName string
 
+	// AuditFields is a map of fields to be included in audit logs
+	AuditFields map[string]any
```

Comment:
I think we should rename this to `RequestAdditionalAuditFields` or similar, so it's obvious it's not the full set of fields (and to remind ourselves that this is injected data that cannot be used to overwrite existing fields).